packages/web-runtime/src/router/index.js

```diff
@@ -93,4 +94,35 @@ router.beforeEach(function(to, from, next) {
   }
 })
 
-export default router
+// type: patch
+// temporary patch till we have upgraded web to the latest vue router which make this obsolete
+// this takes care that routes like 'foo/bar/baz' which by default would be converted to 'foo%2Fbar%2Fbaz' stay as they are
+// should immediately go away and be removed after finalizing the update
+// to apply the patch to a route add meta.patchCleanPath = true to it
+// to patch needs to be enabled on a route level, to do so add meta.patchCleanPath = true property to the route
+const patchRouter = router => {
+  const bindMatcher = router.match.bind(router)
+  const cleanPath = route =>
+    [
+      ['%2F', '/'],
+      ['//', '/']
+    ].reduce((path, rule) => path.replaceAll(rule[0], rule[1]), route || '')
+
+  router.match = (raw, current, redirectFrom) => {
+    const bindMatch = bindMatcher(raw, current, redirectFrom)
+
+    if (!get(bindMatch, 'meta.patchCleanPath', false)) {
+      return bindMatch
+    }
+
+    return {
+      ...bindMatch,
+      path: cleanPath(bindMatch.path),
+      fullPath: cleanPath(bindMatch.fullPath)
+    }
+  }
+
+  return router
+}
+
+export default patchRouter(router)
```

Comment:
Please add a comment that the result of this call is a patched, otherwise fully functional router. Keep the name `patchRouter` like it is. Thought about it again, I like it better than something like `patchedRouter`. Name is good.